### PR TITLE
replaced react-error-boundry with react router error handling.

### PR DESCRIPTION
**Pull request by btirth**
Replaced react-error-boundary with React Router error handling #19307

### CODE REVIEW

File: js/apps/admin-ui/src/App.tsx [hunk 1]
```diff
@@ -2,15 +2,13 @@ import type KeycloakAdminClient from "@keycloak/keycloak-admin-client";
 import { Page } from "@patternfly/react-core";
 import type Keycloak from "keycloak-js";
 import { PropsWithChildren, Suspense } from "react";
-import { ErrorBoundary } from "react-error-boundary";
```
Comment (jonkoops) [inline js/apps/admin-ui/src/App.tsx:5]: If this is the only place this is package was being used I believe we can remove it from the package file.

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: js/apps/admin-ui/src/App.tsx (hunk 1 above)
Comment (tirth1) [reply js/apps/admin-ui/src/App.tsx:5]: we are using useErrorHandler in AdminClient.tsx

Comment (tirth1) [reply js/apps/admin-ui/src/App.tsx:5]: useErrorHandler from react-error-boundary is being use in useFetch inside AdminClient.tsx. Should I log the error or throw the error if I remove the useErrorHandler?

[https://github.com/keycloak/keycloak/blob/main/js/apps/admin-ui/src/context/auth/AdminClient.tsx](url)

**Comment by tirth1** [comment]
useErrorHandler from reach-error-boundry was being used in useFetch in AdminClient. SInce now useFetch is moved in separate file I have removed it from there and package file.

**Comment by jonkoops** [comment]
I'm currently bogged down with some other tasks so it might take a bit for me to get to reviewing this.

**Comment by jonkoops** [comment]
Did a quick rebase to ensure the lockfile has been moved to PNPM. Will do a review shortly.

**Comment by jonkoops** [comment]
Upon closer inspection I see we cannot handle all errors in React Router. The reason for this is that if an error is thrown in an effect it does not actually bubble up to the error boundary. This is also the reason we are need to manually call [`onError`](https://github.com/bvaughn/react-error-boundary/tree/v3.1.4#onerror).

@tirth1 thanks for picking this up, but I don't see a good solution to incorporate React Router at this moment. Apologies for the poor specs.

**Comment by tirth1** [comment]
hi @jonkoops, an error is thrown in an effect it does not actually bubble up to the error boundary: Is this the only issue here? I may have found the workaround for this.

**Comment by jonkoops** [comment]
> Is this the only issue here?

Yes, pretty much. It has to do with how React handles errors, so there is very little we can do about it.

> As per my findings we can use useState for error and set the error in useEffect instead of throwing it. And if an error is present we throw an error.

Not entirely sure what you mean here. AFAIK this would still not allow us to bubble the error during a phase where React is still reconciling the components. Do you have an example of this?

**Comment by tirth1** [comment]
I am talking about something like this.
```
  const [error, setError] = useState<Error | undefined>(undefined);

  if(error) {
    throw error;
  }

  useEffect(() => {
      //code
      .then((result) => (
             // code
      )
      .catch((error) => {
          setError(error);
      });
    // code
  }, []);
```

**Comment by jonkoops** [comment]
Hmmm, yeah this could work perhaps. But it feels like another layer indirection that might complicate things. Perhaps it is better to keep both libraries around for now. I would like to upgrade to the latest version of `react-error-boundary`.

**Comment by jonkoops** [comment]
@tirth1 see #20043

**Pull request closed without merging**